pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -972,4 +1018,31 @@ public void onUpdate(String path, Map<String, String> data, Stat stat) {
         return dynamicConfigurationMap;
     }
 
-}
+    private void processPendingLoadTopic() {
+        lock.lock();
```

Comment:
it is to prevent race condition which may cause topic reside in `pendingTopicLoadingQueue` forever.
eg:
- if `MaxConcurrentTopicLoadRequest`=1 and if two topics (topic1, topic2) come at same time then one of them (topic2) will be added into pendingQueue. 
- as soon as topic1 completes: it tries to release semaphore and `poll` topic2 from queue to createTopic again, but if 3rd new topic3 acquired a permit before it and when it releases it may find `pendingTopicLoadingQueue` empty as we have already `poll` topic2 it before. after that topic2 will be added again on [line-1037](https://github.com/rdhabalia/pulsar/blob/406b0fc6b8efcc1451e163858a1e3f72860968fd/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java#L1037).

So, added lock as [pendingTopicLoadingQueue.poll()](https://github.com/rdhabalia/pulsar/blob/406b0fc6b8efcc1451e163858a1e3f72860968fd/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java#L1024) and [pendingTopicLoadingQueue.add(pendingTopic)](https://github.com/rdhabalia/pulsar/blob/406b0fc6b8efcc1451e163858a1e3f72860968fd/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java#L1037) is not atomic.
Do you think we should ignore this situation?